tests/test_uniquediffs.php

```diff
@@ -0,0 +1,148 @@
+<?php
+//
+// After including cdash_test_case.php, subsequent require_once calls are
+// relative to the top of the CDash source tree
+//
+require_once dirname(__FILE__) . '/cdash_test_case.php';
+require_once 'include/common.php';
+require_once 'include/pdo.php';
+
+class UniqueDiffsTestCase extends KWWebTestCase
+{
+    public function __construct()
+    {
+        parent::__construct();
+        $this->BuildId = null;
+    }
+
+    public function testUniqueDiffs()
+    {
+        // Find the highest buildid created so far and add one to it.
+        // This should give us a buildid that doesn't exist yet.
+        $pdo = get_link_identifier()->getPdo();
+        $stmt = $pdo->query('SELECT id FROM build ORDER BY id DESC LIMIT 1');
+        $row = $stmt->fetch();
+        $buildid = $row['id'];
+        $buildid++;
+
+        // Verify that this buildid does not exist yet.
+        $safe_build = false;
+        while (!$safe_build) {
+            $stmt = $pdo->prepare('SELECT id FROM build WHERE id=?');
+            $stmt->execute([$buildid]);
+            $row = $stmt->fetch();
+            if (!$row) {
+                $this->BuildId = $buildid;
+                $safe_build = true;
+            } else {
+                $buildid++;
+            }
+        }
+
+        // Perform multiple INSERTs that violate the unique constraint
+        // and verify that only one row is recorded.
+        //
+        // builderrordiff
+        $stmt = $pdo->prepare(
```

Comment:
I think checking the row count may not be the best way to test this since an insert can fail for reasons other than the one we want (unique constraint violation). I think `PDO::errorCode` can get us what we want, the only annoyance here is comparing against the relevant MySQL *and* Postgres error codes.

This should probably provoke some thought on the future of PDO in CDash and if we eventually want to migrate to PDO throwing exceptions on errors.